### ISSUE TITLE
Increase semantic markup for table component

### ISSUE THREAD

**Issue by 1Copenut** [CMSgov/design-system#424]
**Description**
I'd like to propose adding additional heading rows, captions, and footers to our [data table component](https://design.cms.gov/components/table/). These items can help users understand layered or dense data concepts. Attaching a screenshot of a fictitious baseball table I wrote for a tech demonstration. All code is available for review and as guidance when converting to React.

---
![Screen Shot 2019-05-22 at 11 57 37 AM](https://user-images.githubusercontent.com/934879/58194051-161f6c80-7c8a-11e9-917c-d5681ef82fda.png)

**Comment by line47** [comment]
Hey @1Copenut Can you provide a bit more context here - are you wanting a more complicated table to be added than the current `Table with two headers and a caption` ?

**Comment by 1Copenut** [comment]
@line47 Yes, I think it's worth doing to have a more complicated table structure illustrated. The table I included in the issue description does a lot behind the scenes to expand the abbreviations to screen readers, and associate each `<td>` cell with both heading rows using IDs and `scope` attributes. We may not have a use for something this complex yet, but I believe we will going forward.

I posted a [public codepen](https://codepen.io/tpierce_402/pen/YzzqwmZ) with the relevant code for review.
